File: Trees/invertTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
package Trees;

import javax.swing.tree.TreeNode;

class invertTree {
    public TreeNode invertTree(TreeNode root) {
        //1. Define the base case
        if(root == null){
            return null;
        }

        //2. Define the left and right of the root
        TreeNode left = this.invertTree(root.left);
        TreeNode right = this.invertTree(root.right);

        //3. swap left and right for the root
        root.left = right;
        root.right = left;

        //4. return the root
        return root;
    }
}

File: Python_Solns/Trees/invertTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        #1. base case
        if(root == None):
            return None
        
        #2. Get the left and right children
        right = self.invertTree(root.right)
        left = self.invertTree(root.left)

        #3. reverse the left and right children
        root.left = right
        root.right = left

        #4. return the root
        return root

        